kore/src/Kore/Repl/Data.hs

```diff
@@ -388,36 +393,80 @@ initializeProofFor
     :: MonadState (ReplState claim) m
     => Claim claim
     => claim
+    -> ClaimIndex
     -> m ()
-initializeProofFor claim =
+initializeProofFor claim cindex = do
+    gphs <- gets graphs
+    lbls <- gets labels
     modify (\st -> st
-        { graph  = emptyExecutionGraph claim
+        { graphs  = Map.insert cindex (emptyExecutionGraph claim) gphs
         , claim  = claim
+        , claimIndex = cindex
         , node   = ReplNode 0
-        , labels = Map.empty
+        , labels = Map.insert cindex Map.empty lbls
+        })
+
+switchToProof
+    :: MonadState (ReplState claim) m
+    => Claim claim
+    => claim
+    -> ClaimIndex
+    -> m ()
+switchToProof claim cindex = do
```

Comment:
Wouldn't it make more sense for this function to make sure the map for labels and the new execution graph are initialized, rather than their respective getters?